subsys/shell/shell_retval.c

```diff
@@ -0,0 +1,10 @@
+#include <zephyr/shell/shell.h>
+
+static int cmd_get_retval(const struct shell *shell, size_t argc, char **argv){
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+	shell_print(shell,"%d",shell->ctx->return_code);
+	return 0;
+}
+
+SHELL_CMD_REGISTER(return_code, NULL, "Gets return code of previous command.", cmd_get_retval);
```

Comment:
You don't need a new file. If it depends on `SHELL_CMDS` then you can add it to `shell_cmds.c` and use `SHELL_COND_CMD_REGISTER` (or better `SHELL_COND_CMD_ARG_REGISTER` and indicate that there is only 1 fixed argument).